File: q0997/q0997.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        # Tracking the how many people is one person trusted by
        trustedCount = [0 for _ in range(n+1)]
        # Tracking how many people one person is willing to trust
        trustingCount = [0 for _ in range(n+1)]
        for pair in trust: 
            trustedCount[pair[1]] += 1
            trustingCount[pair[0]] += 1
        
        # Trusted by everyone else (n-1) and
        # Not trusting anyone else 0
        for i in range(1, n+1):
            if trustedCount[i] == n-1 and trustingCount[i] == 0: 
                return i
        return -1

File: q0997/q0997.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        // Tracking the how many people is one person trusted by
        int[] trustedCount = new int[n + 1];
        // Tracking how many people one person is willing to trust
        int[] trustingCount = new int[n + 1];
        for (int[] pair : trust) {
            trustedCount[pair[1]]++;
            trustingCount[pair[0]]++;
        }

        // Trusted by everyone else (n-1) &&
        // Not trusting anyone else 0
        for (int i = 1; i < n + 1; i++) {
            if (trustedCount[i] == n - 1 && trustingCount[i] == 0)
                return i;
        }
        return -1;
    }
}